### PR TITLE
add read/write data services

### PR DESCRIPTION
**Pull request by AlexMoroz**

### CODE REVIEW

File: docs/ScopeContext/3_public_interfaces.rst [hunk 1]
```diff
@@ -10,14 +10,14 @@ Interfaces to other modules will be realized using ROS (rosjava). Currently 2 in
 	# argument: String payload
 	# returns: Bool success
 
-	rosservice call /write_qr
+	rosservice call /roboy/cognition/memory/data/write
```
Comment (missxa) [inline docs/ScopeContext/3_public_interfaces.rst]: what are the arguments for this service call?

also mention that DataQuery.srv is inside roboy_communication_cognition

File: docs/Usage/1_getting_started.rst [hunk 2]
```diff
@@ -69,5 +69,18 @@ If you are using roboy_memory package in remote mode, please use the username an
...
+
+ROS Configuration
+---------------------------------------------------
+
+ROS interfaces require network setup. (more info: http://wiki.ros.org/ROS/NetworkSetup )
+
+For this two variables in Config class (util package) should be changed.
+
+    ROS_MASTER_URI - defines an URI of roscore module in the network, e.g. "http://bot.roboy.org:11311/"
+    ROS_HOSTNAME - defines the IP address of the machine with rosjava mudule in the network, e.g. "192.168.1.1"
```
Comment (missxa) [inline docs/Usage/1_getting_started.rst]: rosjava module has to be running on the bot.roboy.org too, so one doesn't need to build/run it separately

File: src/main/roboy/memory/services/ReadData.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * ROS Service for saving data object to DB. Data is received as JSON object.
+ * JSON object is parsed using Parser (not implemented) and saved to neo4j.
+ */
+public class ReadData extends AbstractNodeMain {
+    private static String name = "/roboy/cognition/memory/data/read";
+    private Parser parser = new Parser();
+
+    public static void publish(NodeConfiguration nodeConfiguration, NodeMainExecutor nodeMainExecutor) {
```
Comment (missxa) [inline src/main/roboy/memory/services/ReadData.java]: what does it publish?
it starts the node

File: src/main/roboy/memory/services/ReadData.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import roboy_communication_cognition.DataQuery;
+import roboy_communication_cognition.DataQueryRequest;
+import roboy_communication_cognition.DataQueryResponse;
+
+
+/**
+ * ROS Service for saving data object to DB. Data is received as JSON object.
+ * JSON object is parsed using Parser (not implemented) and saved to neo4j.
+ */
+public class ReadData extends AbstractNodeMain {
```
Comment (missxa) [inline src/main/roboy/memory/services/ReadData.java]: so the idea to open a new node for each service? that does not make much sense for ROS. 
optimal approach 
- node: roboy_memory (omit "/" in the node names)
       * service /roboy/cognition/memory/data/read
       * service /roboy/cognition/memory/data/wrtite 
       * ...

File: src/main/roboy/memory/services/WriteData.java [hunk 5]
```diff
@@ -10,27 +11,32 @@
 
 
 /**
- * Save data to DB
+ * ROS Service for saving data object to DB. Data is received as JSON object.
+ * JSON object is parsed using Parser (not implemented) and saved to neo4j.
  */
-public class SaveObject extends AbstractNodeMain {
-    private static String name = "/roboy/cognition/memory/data/save";
+public class WriteData extends AbstractNodeMain {
```
Comment (missxa) [inline src/main/roboy/memory/services/WriteData.java]: again, you open a new node for 1 service

**Comment by Waguramu** [comment]
We will discuss the requested changes today after the main meeting

File: docs/ScopeContext/3_public_interfaces.rst (hunk 1 above)
Comment (AlexMoroz) [inline docs/ScopeContext/3_public_interfaces.rst]: screen shot 2017-07-01 at 17 46 06

**Pull request closed without merging**